core/src/main/scala/cats/Alternative.scala

```diff
@@ -22,6 +22,18 @@ import simulacrum.typeclass
     (as, bs)
   }
 
+  def mapSeparate[A, B, C](fa: F[A])(f: A => Either[B, C])(implicit F: Foldable[F], A: Alternative[F]): (F[B], F[C]) = {
+    import cats.instances.tuple._
+
+    implicit val mb: Monoid[F[B]] = Alternative[F].algebra[B]
+    implicit val mc: Monoid[F[C]] = Alternative[F].algebra[C]
+
+    F.foldMap(fa)(a => f(a) match {
+      case Right(c) => (empty[B], pure(c))
+      case Left(b) => (pure(b), empty[C])
+    })
+  }
+
```

Comment:
I think @peterneyens was suggesting that 
you can just write 
```scala
  import cats.instances.either._
  def mapSeparate[A, B, C](fa: F[A])(f: A => Either[B, C])(implicit FM: Monad[F]): (F[B], F[C]) =
    separate(map(fa)(f))
```
So instead of requiring `F` to be `Foldable`, it requires to be a `Monad`. 
Or we can use your implementation, which requires `Foldable`, which might be more performant? (traverse once instead three times?)

I still think we should name it `partitionEither` or something, the idea is to provide a familiar api to users. `map` and then `separate` probably don't justify a special delegate method also named `mapSeparate`.

Reply:
I think I prefer the `Foldable` way, for the reasons you mentioned. I'm not married to the name, but it makes sense in the same way `flatMap` makes sense, I guess. 

I think I'd call the method on `Foldable` `partitionEither` as you suggested and renamed the one on `Reducible` to `partitionIor` Maybe? 
`partitionE` isn't a great name IMO, because there's no real precedence for the "E" and I think if we want to stay a beginner friendly library we should have as little of these one-letter-abbreviations as possible.

Reply:
The current implementation is better than `map` + `separate` indeed. Just wanted to point out the similarity.

`separate` is available for all `Bifoldable`s while we currently have to use `Either` with `mapSeparate`:
- With `separate` you could also split `Ior` and `Tuple2` (similar to `unzip`).
-  `mapSeparate` + `toEither` will probably be faster than `map(toEither).separate` as you fold once instead of twice.

I agree that `partitionE` isn't a great name.

Reply:
my 2 cents 
1. since we are going to use `Foldable`, it seems to make more sense to be on `Foldable` than `Alternative`.  Having a pair of partition methods on `Foldable` and `Reducible` is what I was looking for. 
1.  both methods should be named `partitionEitherXX` since they both use a function that returns an Either for the actual partition logic. Maybe we can call `Reducible#partitionEitherNE` and `Foldable#partitionEither`
1. maybe we should mention the `map` + `separate` option in the scaladoc in `Foldable#partitionEither`, I think it might worth some promotion.

Reply:
I'm +1 on renaming to `Foldable#partitionEither` and mentioning the `map` + `separate` option in the scaladoc. Disagree on `partitionEitherNE` though, for similar reasons as `partitionE`. 
How about `nonEmptyPartition`? Would go in line with `nonEmptyIntercalate` etc. 🤔

Reply:
I am fine with nonEmptyPartition

Reply:
Renamed to `Foldable#partitionEither` and `Reducible#nonEmptyPartition` :)